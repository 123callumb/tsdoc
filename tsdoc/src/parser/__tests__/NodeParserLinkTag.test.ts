import { TestHelpers } from './TestHelpers';

test('00 Link tags with URL: positive examples', () => {
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * {@link http://example1.com}',
    ' * {@link mail:bob@example2.com}',
    ' * {@link http://example3.com|link text}',
    ' * {@link http://example4.com|}',
    ' * {@link http://example5.com|',
    ' *}',
    ' */'
  ].join('\n'));
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * {@link    mail:bob@example6.com   |  link text }',
    ' * {@link  ',
    ' *   http://example7.com  ',
    ' *   | ',
    ' *   link text ',
    ' * }',
    ' */'
  ].join('\n'));
});
