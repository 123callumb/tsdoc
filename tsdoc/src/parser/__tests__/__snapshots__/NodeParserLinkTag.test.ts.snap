// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`00 Link tags with URL: positive examples 1`] = `
Object {
  "buffer": "/**[n] * {@link http://example1.com}[n] * {@link mail:bob@example2.com}[n] * {@link http://example3.com|link text}[n] * {@link http://example4.com|}[n] * {@link http://example5.com|[n] *}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@link http://example1.com}",
    "{@link mail:bob@example2.com}",
    "{@link http://example3.com|link text}",
    "{@link http://example4.com|}",
    "{@link http://example5.com|",
    "}",
  ],
  "logMessages": Array [],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": " ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "http://example1.com",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": " ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "mail:bob@example2.com",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": " ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "http://example3.com",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "|",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "link text",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": " ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "http://example4.com",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "|",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": " ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "http://example5.com",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "|",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "[n]",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
            ],
          },
        ],
      },
    ],
  },
}
`;

exports[`00 Link tags with URL: positive examples 2`] = `
Object {
  "buffer": "/**[n] * {@link    mail:bob@example6.com   |  link text }[n] * {@link  [n] *   http://example7.com  [n] *   | [n] *   link text [n] * }[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@link    mail:bob@example6.com   |  link text }",
    "{@link",
    "  http://example7.com",
    "  |",
    "  link text",
    "}",
  ],
  "logMessages": Array [],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": "    ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "mail:bob@example6.com",
                    "nodeSpacing": "   ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "|",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "  link text ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
              Object {
                "kind": "LinkTag",
                "nodes": Array [
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "@link",
                    "nodeSpacing": "[n]  ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "http://example7.com",
                    "nodeSpacing": "[n]  ",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "|",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "[n]  link text[n]",
                  },
                  Object {
                    "kind": "Particle",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodeExcerpt": "[n]",
              },
            ],
          },
        ],
      },
    ],
  },
}
`;
